src/psyclone/psyir/symbols/symbol_table.py

```diff
@@ -546,6 +546,182 @@ def add(self, new_symbol, tag=None):
+        this table. If a symbol with the same name appears in both tables but
+        is imported from different Containers then the two are returned.
+
+        :returns: the first two Symbols found with an import clash, if any.
+        :rtype: Optional[Tuple[:py:class:`psyclone.psyir.symbols.Symbol`, \
+                               :py:class:`psyclone.psyir.symbols.Symbol`]]
+        '''
+        other_import_names = [sym.name for sym in other_table.imported_symbols]
+        for sym in self.imported_symbols:
+            if sym.name in other_import_names:
+                routine_sym = other_table.lookup(sym.name)
+                if (routine_sym.interface.container_symbol.name !=
+                        sym.interface.container_symbol.name):
+                    return sym, routine_sym
+        return None
+
+    def _add_container_symbols_from_table(self, other_table):
+        '''
+        Takes container symbols from the supplied symbol table and adds them to
+        this table. All references to each container symbol are also updated.
+        (This is a preliminary step to adding all symbols from other_table to
+        this table.)
+
+        :param table: the symbol table at the call site.
+        :type table: :py:class:`psyclone.psyir.symbols.SymbolTable`
+        :param routine_table: the symbol table of the routine being inlined.
+        :type routine_table: :py:class:`psyclone.psyir.symbols.SymbolTable`
+
+        '''
+        for csym in other_table.containersymbols:
+            if csym.name in self:
+                # We have a clash with another symbol in this table.
+                self_csym = self.lookup(csym.name)
+                if not isinstance(self_csym, ContainerSymbol):
+                    # The symbol in this table is not a Container so we
+                    # can rename it.
+                    self.rename_symbol(
+                            self_csym,
+                            self.next_available_name(
+                                csym.name, other_table=other_table))
+                    # We can then add an import from the Container.
+                    self.add(csym)
+                else:
+                    # If there is a wildcard import from this container then
+                    # we'll need that in this Table too.
+                    if csym.wildcard_import:
+                        self_csym.wildcard_import = True
+            else:
+                self.add(csym)
+            # We must update all references to this ContainerSymbol
+            # so that they point to the one in this table instead.
+            imported_syms = other_table.symbols_imported_from(csym)
+            for isym in imported_syms:
+                if isym.name in self:
+                    # We have a potential clash with a symbol imported
+                    # into the other table.
+                    callsite_sym = self.lookup(isym.name)
```

Comment:
Rename callsite to something more generic

Reply:
Oops, done.